### DISCUSSION TITLE
ERROR:TeleBot:'AsyncTeleBot' object has no attribute 'register_next_step_handler'

### DISCUSSION THREAD

**Discussion by hythem1979** [eternnoir/pyTelegramBotAPI#1670] [Q&A] [answered]
Hello, I ported my code to AsyncTeleBot.

in this line of code
await bot.register_next_step_handler(msg, process_email_step)

I get this error
ERROR:TeleBot:'AsyncTeleBot' object has no attribute 'register_next_step_handler'

searched documentation, samples and web couldn't find a solution or a start.

**Comment by coder2020official** [comment, marked as answer]
Unfortunately, async version does not support register next step handler, and it won't support it. Use states instead.

**Reply by hythem1979** [reply]
thank you for replay but how to use status instead 
to handle the same scenario 
the user send command
the bot ask question and wait for user's answer
the user send the answer
the bot verify the answer and accept or repeat question again

**Reply by coder2020official** [reply]
https://github.com/eternnoir/pyTelegramBotAPI/blob/master/examples/custom_states.py

**Reply by hythem1979** [reply]
I followed this sample
https://github.com/eternnoir/pyTelegramBotAPI/blob/master/examples/asynchronous_telebot/custom_states.py

it works fine with general custom filters but win I use state filter it did not make anything at all

**Reply by coder2020official** [reply]
we cannot guess your code

**Reply by hythem1979** [reply]
yes you're right but my code is lot missy 😂😂 I could not extract the target code, and I found the problem.
in the first argument in set_state I converted to (string)
id = str(message.from_user.id)
await bot.set_state(id, MyStates.user_email, message.chat.id)

changed to:
await bot.set_state(message.from_user.id, MyStates.user_email, message.chat.id)

thanks for help
